client.go

```diff
@@ -327,7 +327,12 @@ func (c *Client) acquireLock(ctx context.Context, opt *acquireLockOptions) (*Loc
 			return l, nil
 		}
 		c.logger.Println("Sleeping for a refresh period of ", getLockOptions.refreshPeriodDuration)
-		time.Sleep(getLockOptions.refreshPeriodDuration)
+		select {
+		case <-ctx.Done():
+			break
+		case <-time.After(getLockOptions.refreshPeriodDuration):
+			continue
```

Comment:
no need to declare `continue` here.

Reply:
Just a bare case with no statement following it?  I wasn't sure if that was idiomatic.  Ok.